rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,761 @@
+    parameters:
+      - name: namespace
+        in: path
+        required: true
+        schema:
+          type: string
+    get:
+      tags:
+        - Catalog API
+      summary: Load the metadata properties for a namespace
+      operationId: loadNamespaceMetadata
+      description: Return all stored properties for a given namespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+        '417':
+          description: Namespace not found
+          # TODO - Make a canonical example for NamespaceNotFound
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchNamespaceError'
+              example:
+                '{ error: { message: "Namespace does not exist", type: "NoSuchNamespaceException", code: 41701 }'
+    put:
+      tags:
+        - Catalog API
+      summary: Add or overwrite properties to an existing namespace
+      operationId: setNamespaceProperties
+      description:
+        Adds propertiess for a namespace. This will overwrite any existing properties,
+        and merge with the others.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RemovePropertiesRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            'application/json':
+              schema:
+                type: boolean
+        '417':
+          description: Namespace not found
+          # TODO - Make a canonical example for NamespaceNotFound
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchNamespaceError'
+              example:
+                '{ error: { message: "Namespace does not exist", type: "NoSuchNamespaceException", code: 41701 }'
+        '409':
+          description: Namespace already exists
+          content:
+            applicaton/json:
+              schema:
+                $ref: '#/components/schemas/NamespaceAlreadyExistsError'
+              example:
+                '{ error: { message: "Namespace already exists", type: "AlreadyExistsException", code: 40901 }'
+    post:
+      tags:
+        - Catalog API
+      summary: Overwrite a namespace's properties with a new set of properties
+      description: Set properties on a namespace
+      operationId: setProperties
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SetPropertiesRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                example: '{ data: { success: true }, error: { } }'
+  # TODO - Pagination
+  /v1/namespaces/list:
```

Comment:
I think this should be `/v1/namespaces`? Why the additional `/list`?

Reply:
I believe you are right. There's not a need for that. I was working across a few copies and missed that `/v1/namespaces` isn't taken now, just `/v1/namespaces/{namespace}`.

I agree `/list` is likely not needed.

Reply:
Ok I removed it. For listing namespaces, it's currently

`GET /v1/namespaces[?namespacce=$ns]` where the query parameter `namespace` is optional. This is to allow for users to list all top level namespaces or to list namespaces contained within a given namespace.

If we prefer to have `/v1/namespaces` and `/v1/namespaces/{namespace}`, I'll have to write them up as two different routes, but it is doable. Please let me know what you think.

I don't currently like the inconsistency of having `namespace` be a query parameter here, but as a path parameter it's required.

Reply:
I think `GET /v1/namespaces[?parent=$parent]` could work as a generic LIST namespace API endpoint, what do you think?

Reply:
Oh I like that much more. Avoiding the use of `namespace` as a query parameter when it's a path parameter everywhere else is likely to be a lot less confusing.

I'll update the route and the description to clarify. Thanks Jack!

Reply:
This has been changed.